File: backend/app/constants/roles.py
# File: app/constants/roles.py
from enum import Enum


class RoleCode(Enum):
    """Mã vai trò chuẩn dùng xuyên suốt toàn hệ thống."""

    QTV = "QTV"
    BQL = "BQL"
    TP_KDTC = "TP_KDTC"
    NV_KDTC = "NV_KDTC"
    TP_VHBT = "TP_VHBT"
    NV_VHBT = "NV_VHBT"
    KHACH_THUE = "KHACH_THUE"


INTERNAL_STAFF_ROLES = {
    RoleCode.QTV,
    RoleCode.BQL,
    RoleCode.TP_KDTC,
    RoleCode.NV_KDTC,
    RoleCode.TP_VHBT,
    RoleCode.NV_VHBT,
}

KDTC_ROLES = {
    RoleCode.TP_KDTC,
    RoleCode.NV_KDTC,
}

VHBT_ROLES = {
    RoleCode.TP_VHBT,
    RoleCode.NV_VHBT,
}

MANAGER_ROLES = {
    RoleCode.BQL,
    RoleCode.TP_KDTC,
    RoleCode.TP_VHBT,
}

ACCOUNT_ADMIN_ROLES = {
    RoleCode.QTV,
}

AUDIT_VIEW_ROLES = {
    RoleCode.QTV,
    RoleCode.BQL,
}

CONTRACT_LIST_VIEW_ROLES = {
    RoleCode.BQL,
    RoleCode.TP_KDTC,
    RoleCode.NV_KDTC,
}

PREMISE_VIEW_ROLES = {
    RoleCode.BQL,
    RoleCode.TP_VHBT,
    RoleCode.NV_VHBT,
    RoleCode.KHACH_THUE,
}

File: frontend/src/constants/roles.js
export const ROLE = {
  QTV: "QTV",
  BQL: "BQL",
  TP_KDTC: "TP_KDTC",
  NV_KDTC: "NV_KDTC",
  TP_VHBT: "TP_VHBT",
  NV_VHBT: "NV_VHBT",
  KHACH_THUE: "KHACH_THUE",
};

export const ROLE_LABEL = {
  [ROLE.QTV]: "Quản trị viên",
  [ROLE.BQL]: "Ban Quản Lý",
  [ROLE.TP_KDTC]: "Trưởng phòng Kinh doanh - Tài chính",
  [ROLE.NV_KDTC]: "Nhân viên Kinh doanh - Tài chính",
  [ROLE.TP_VHBT]: "Trưởng phòng Vận hành - Bảo trì",
  [ROLE.NV_VHBT]: "Nhân viên Vận hành - Bảo trì",
  [ROLE.KHACH_THUE]: "Khách thuê",
};

export const INTERNAL_ROLES = [
  ROLE.QTV,
  ROLE.BQL,
  ROLE.TP_KDTC,
  ROLE.NV_KDTC,
  ROLE.TP_VHBT,
  ROLE.NV_VHBT,
];

export const KDTC_ROLES = [ROLE.TP_KDTC, ROLE.NV_KDTC];
export const VHBT_ROLES = [ROLE.TP_VHBT, ROLE.NV_VHBT];
export const MANAGER_ROLES = [ROLE.BQL, ROLE.TP_KDTC, ROLE.TP_VHBT];
